src/metrics.ts

```diff
@@ -177,13 +183,18 @@ export const MetricsProcessor = (
   };
 
   const _send = (): void => {
+    if (closed) {
+      return;
+    }
+
     const metrics: Metrics = _summarize();
     if (metrics) {
       log.debug('Start sending metrics data');
       api
         .postMetrics(environment, cluster, metrics)
         .then((response) => {
           log.debug('Metrics server returns: ', response.status);
```

Comment:
Should all these older messages be killed off now that we have the standardised ones?

Reply:
It's a good point. We haven't done that with Python or Ruby. We probably should. It will require a bit of time which I don't have right now - if it's ok with you, I'll opt to leave the old ones in, and get them removed when I next have time.